### PR TITLE
METRON-906: Rest service storm configuration does not allow for proper URLs

### PR DESCRIPTION
**Pull request by justinleet**
## Contributor Comments
Allows for protocol to be used in the Storm REST URL (e.g. http:// and https://).  Prior to this, http:// was automatically prepended.  In the event that no protocol is given, http:// will be prepended for backwards compatibility.

Unit tests are added to ensure that the protocol is handled properly. Full dev was spun up to ensure that things still functioned as expected, with the protocol provided.

## Pull Request Checklist

Thank you for submitting a contribution to Apache Metron.  
Please refer to our [Development Guidelines](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=61332235) for the complete guide to follow for contributions.  
Please refer also to our [Build Verification Guidelines](https://cwiki.apache.org/confluence/display/METRON/Verifying+Builds?show-miniview) for complete smoke testing guides.  

In order to streamline the review of the contribution we ask you follow these guidelines and ask you to double check the following:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? If not one needs to be created at [Metron Jira](https://issues.apache.org/jira/browse/METRON/?selectedTab=com.atlassian.jira.jira-projects-plugin:summary-panel). 
- [x] Does your PR title start with METRON-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.
- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?

### For code changes:
- [x] Have you included steps to reproduce the behavior or problem that is being changed or addressed?
- [x] Have you included steps or a guide to how the change may be verified and tested manually?
- [x] Have you ensured that the full suite of tests and checks have been executed in the root incubating-metron folder via:
  ```
  mvn -q clean integration-test install && build_utils/verify_licenses.sh 
  ```

- [x] Have you written or updated unit tests and or integration tests to verify your changes?
- [x] Have you verified the basic functionality of the build by building and running locally with Vagrant full-dev environment or the equivalent?

#### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.
It is also recommended that [travis-ci](https://travis-ci.org) is set up for your personal repository such that your branches are built there before submitting a pull request.

### CODE REVIEW

File: metron-deployment/roles/ambari_config/vars/single_node_vm.yml [hunk 1]
```diff
@@ -102,7 +102,7 @@ configurations:
 
 required_configurations:
   - metron-env:
-      storm_rest_addr: "{{ groups.ambari_slave[0] }}:8744"
+      storm_rest_addr: "http://{{ groups.ambari_slave[0] }}:8744"
```
Comment (cestella) [inline metron-deployment/roles/ambari_config/vars/single_node_vm.yml]: Could you explain why we are presuming `http://` here

Comment (justinleet) [inline metron-deployment/roles/ambari_config/vars/single_node_vm.yml]: It probably shouldn't be.  I did that before the backwards compatibility commit.  It'll be assumed at the code level if no protocol is there anyway, so do we have a preference on explicit vs. implicit?

It seems more like I need documentation around changing that parameter when storm https is enabled (both non-Ambari and Ambari need it).

Comment (dlyle65535) [inline metron-deployment/roles/ambari_config/vars/single_node_vm.yml]: You'll need to change a ton more than that parameter to enable https. Remember, this file defines a specific cluster. Unsecure with no https. Either http or omitted is fine.

Comment (justinleet) [inline metron-deployment/roles/ambari_config/vars/single_node_vm.yml]: @cestella Thoughts?  @dlyle65535 is right that it's the definition of a specific cluster.  I'd argue that it's the user's responsibility to update anything afterwards when they set up SSL (e.g. through Ambari).  All I was thinking of documenting was just to ensure that field gets updated, but I'm not sure how useful that is anyway.

Comment (cestella) [inline metron-deployment/roles/ambari_config/vars/single_node_vm.yml]: That's fine; I get it.

**Comment by simonellistonball** [comment]
Is there anything preventing this getting merged?

**Comment by cestella** [comment]
+1

**Comment by ottobackwards** [comment]
IS someone merging this?  Or shall I?

**Pull request closed without merging**